### ISSUE TITLE
is the android app actually downloadable?

### ISSUE THREAD

**Issue by vesper8** [jlooper/elocute#1]
Hi there!

I saw your post on the vue-nativescript repo, there are very few sample apps out there and I think playing with a working app is a great way to learn

I checked out the https://elocute.me/ website and tried to click on the 'download for android' link but nothing happens. I was just wondering if I want to test out the android companion app do I need to run it in a simulator or is it already on the app store?

**Comment by jlooper** [comment]
Hi, these apps aren’t quite ready for prime time :) - they need to be really good if I’m hoping teachers and students will use them. I’ve also mostly tested on iOS. But you can find the code in the mobile folder if you want to try them on device. Instructions to run them are there.

**Comment by vesper8** [comment]
I followed the instructions. Cloned the repo, switched to the mobile folder and ran npm install

started up my android emulator pixel 7.1 on Genymotion

then ran webpack --watch --env.tns --env.android

in one terminal, and in the other terminal ran tns debug android

I'm running node 9.5

and this generated this error

```
An uncaught Exception occurred on "main" thread.
java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: 

Cannot compile /data/data/com.ladeezfirstmedia.elocute/files/app/main.js

SyntaxError: Unexpected token import
File: "file:///data/data/com.ladeezfirstmedia.elocute/files/app/main.js, line: 1, column: 60

StackTrace: 
	Frame: function:'require', file:'', line: 1, column: 266


SyntaxError: Unexpected token import
File: "<unknown>, line: 1, column: 265

StackTrace: 
	Frame: function:'require', file:'', line: 1, column: 266


	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5406)
	at android.app.ActivityThread.-wrap2(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6119)
	at java.lang.reflect.Method.invoke(Native Method
```

any idea what's wrong?

**Comment by jlooper** [comment]
Are you running tns commands within the tns folder? Also I need to debug this a bit more on Android

**Comment by vesper8** [comment]
oh no oops.. I wasn't

Is the webpack --watch --env.tns --env.android command also supposed to be run from the /mobile/tns folder or just /mobile ? It seems to fail if I try to run it from the tns folder

When running tns debug android from inside the tns folder I'm getting a different error now:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processF0F1F2DebugGoogleServices'.
> File google-services.json is missing. The Google Services Plugin cannot function without it.
   Searched Location:
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/src/F0F1F2/debug/google-services.json
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/src/debug/F0F1F2/google-services.json
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/src/F0F1F2/google-services.json
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/src/debug/google-services.json
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/src/F0F1F2Debug/google-services.json
  /DEV/__NATIVESCRIPT/elocute/mobile/tns/platforms/android/google-services.json
```

**Comment by jlooper** [comment]
This is a Firebase dependency. Try tns run instead of debug to load up the files

**Comment by vesper8** [comment]
tried starting all over in case running the commands from the wrong folders was causing the error

```
git clone https://github.com/jlooper/elocute.git
cd elocute/mobile
npm i
webpack --watch --env.tns --env.android
```

in another terminal

```
cd eloctute/mobile/tns
tns run android (also tried just tns run, and tns debug android
```

And it gives the same error as above

**Comment by vesper8** [comment]
grabbing the file in ```elocute/mobile/app/App_Resources/Android/google-services.json```

and moving it to ```elocute/mobile/tns/platforms/android```(which was one of the default locations it was looking for it

has allowed me to get past this error

I then see that it is trying to load the app in the emulator, I can see the splash screen with the cute face.. and then it goes into a crash loop and spits out this error over and over

```
Refreshing application...
JS: 'Logged out from firebase (init\'s onAuthStateChanged callback)'
JS: Error: Css styling failed: Error: undefined:1:16: missing '{'
JS: Missing Image with resourceID: res://seal
JS: Missing Image with resourceID: res://seal
Successfully synced application com.ladeezfirstmedia.elocute on device 192.168.58.101:5555.
Executing before-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/before-prepare/nativescript-dev-android-snapshot.js
Executing before-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/before-prepare/nativescript-plugin-firebase.js
Preparing project...
Project successfully prepared (Android)
Executing after-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/after-prepare/firebase-build-gradle.js
Configure firebase
Executing after-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/after-prepare/firebase-copy-google-services.js
Executing after-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/after-prepare/nativescript-dev-android-snapshot.js
Executing after-prepare hook from /DEV/__NATIVESCRIPT/elocute/mobile/tns/hooks/after-prepare/nativescript-plugin-firebase.js
ActivityManager: Start proc 3837:com.ladeezfirstmedia.elocute/u0a78 for activity com.ladeezfirstmedia.elocute/com.tns.NativeScriptActivity
JS: 'NativeScript-Vue has "Vue.config.silent" set to true, to see output logs set it to false.'
JS: 'LoginOrSignup mounted'
JS: 'LoginOrSignup mounted'
JS: Error: Css styling failed: Error: undefined:1:16: missing '{'
Successfully transferred app.css.
Successfully transferred starter.js.
Successfully transferred package.json.
Successfully transferred package.json.
Successfully transferred starter.js.
Successfully transferred app.css.
Successfully transferred package.json.
Successfully transferred starter.js.
Successfully transferred app.css.
```

**Comment by jlooper** [comment]
There are 2 problems, one is this odd app.css.js file that seems to bork Android, try deleting it. And the other is a missing image, I think I’m missing a seal image in the Android App_Resources folder.

**Comment by jlooper** [comment]
Make sure that the files in App_Resources in root match those on your local tns folder. We don’t use a symlink with this project so have to manually copy, it’s not ideal

**Comment by sivabudati** [comment]
After **npm install**, I tried to run this for android with **tns run android --bundle**. After syncing the app on the emulator, I see an error. Relevant part below. I created a different app from the scratch and was able to use manual routing - no issues there. Do I need to do anything with Webpack configuration before I can run?

`JS: 'LoginOrSignup mounted'
JS: 'firebase.init done'
JS: 'Logged out from firebase (init\'s onAuthStateChanged callback)'
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.callJSMethodNative(Native Method)
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1116)
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.callJSMethodImpl(Runtime.java:996)
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.callJSMethod(Runtime.java:983)
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.callJSMethod(Runtime.java:967)
12-25 10:04:26.008  3975  3975 E AndroidRuntime: 	at com.tns.Runtime.callJSMethod(Runtime.java:959)
System.err: com.tns.NativeScriptException: 
System.err: Calling js method onAuthStateChanged failed
System.err: 
System.err: TypeError: nativescript_vue__WEBPACK_IMPORTED_MODULE_0___default.a.navigateTo is not a function
System.err: File: "file:///data/data/com.ladeezfirstmedia.elocute/files/app/bundle.js, line: 1920, column: 62
System.err: 
`
